theano/tensor/basic.py

```diff
@@ -2312,17 +2315,22 @@ def ones_like(model, dtype=None):
     """
     if dtype is None:
         dtype = model.type.dtype
-    ret = fill(model, constant(1.0, dtype=dtype))
-    return ret
+    ret = constant(1.0, dtype=dtype)
+    if opt and ret.type == model.type:
```

Comment:
I had profiled with this. This causes  a slight slow down. Also, at the end of the profiling there wasn't any error when i removed this change. Do you remember which error this diff fixed ?